src/common/application.cpp

```diff
@@ -49,36 +48,39 @@ namespace
     // loop has begun.
     bool gIsRunning = false;
 
-    void handleSignal(int signal)
+    void handleSignal(const std::error_code& error, int signal)
     {
-        switch (signal)
+        if (!error)
```

Comment:
(I haven't looked at the usage for this)

I dont understand this logic? What's the point of this?

Also, you could invert this condition and bail out early instead of wrapping the whole switch